src/ruby/puppetserver-lib/puppet/server/http_client.rb

```diff
@@ -92,6 +79,21 @@ def get(url, headers={}, options={})
   end
 
   def create_common_request_options(url, headers, options)
+    # If credentials were supplied for HTTP basic auth, add them into the headers.
+    # This is based on the code in lib/puppet/reports/http.rb.
```

Comment:
It looks like this comment was never very correct and is very incorrect at this point. Can you remove it while you're here?